File: leetcode/POTD/2025-02-04/solution.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        max_sum, curr = 0, nums[0]
        for i in range(1, len(nums)):
            if nums[i] > nums[i-1]:
                curr += nums[i]
            else :
                max_sum = max(max_sum, curr)
                curr = nums[i]
        return max(max_sum, curr)

File: leetcode/POTD/2025-02-04/solution.go
func maxAscendingSum(nums []int) int {
    maxSum, curr := 0, nums[0]

    for i := 1; i < len(nums); i++ {
        if nums[i] > nums[i-1] {
            curr += nums[i]
        } else {
            if curr > maxSum {
                maxSum = curr
            }
            curr = nums[i]
        }
    }

    if curr > maxSum {
        maxSum = curr
    }
    
    return maxSum
}